lib/pages/beta/beta.dart

```diff
@@ -0,0 +1,242 @@
+import 'dart:io' show Platform;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:future_loading_dialog/future_loading_dialog.dart';
+import 'package:go_router/go_router.dart';
+import 'package:matrix/matrix.dart';
+import 'package:tawkie/config/app_config.dart';
+import 'package:tawkie/widgets/matrix.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class BetaJoinPage extends StatelessWidget {
+  const BetaJoinPage({super.key});
+
+  void joinBeta() async {
+    try {
+      if (Platform.isIOS) {
+        if (await canLaunchUrl(Uri.parse(AppConfig.iosUrl))) {
+          await launchUrl(Uri.parse(AppConfig.iosUrl));
+        } else {
+          await _launchUrlOrThrow(AppConfig.appleBetaUrl);
+        }
+      } else if (Platform.isAndroid) {
+        if (await canLaunchUrl(Uri.parse(AppConfig.androidUrl))) {
+          await launchUrl(Uri.parse(AppConfig.androidUrl));
+        } else {
+          await _launchUrlOrThrow(AppConfig.androidBetaUrl);
+        }
+      }
+    } catch (e) {
+      if (kDebugMode) {
+        print('Error launching URL: $e');
+      }
+    }
+  }
+
+  Future<void> _launchUrlOrThrow(String url) async {
+    if (await canLaunchUrl(Uri.parse(url))) {
+      await launchUrl(Uri.parse(url));
+    } else {
+      throw 'Could not launch $url';
+    }
+  }
+
+  Future<void> joinGroup({
+    required BuildContext context,
+  }) async {
+    final client = Matrix.of(context).client;
+    const String roomAlias = AppConfig.roomAlias;
+
+    try {
+      if (kDebugMode) {
+        print('Attempting to join room: $roomAlias');
+      }
+
+      // Get room ID from alias
+      final roomAliasResult = await client.getRoomIdByAlias(roomAlias);
+      final roomId = roomAliasResult.roomId;
+
+      if (roomId == null) {
+        final errorMsg = L10n.of(context)!.failedToJoinRoom +
+            L10n.of(context)!.roomIdNullError +
+            roomAlias;
+        if (kDebugMode) {
+          print(errorMsg);
+        }
+        throw Exception(errorMsg);
+      }
+
+      // Check if the user is already a member of the room
+      final room = client.getRoomById(roomId);
+      if (room != null && room.membership == Membership.join) {
+        // Navigate directly to the room
+        context.go('/rooms/$roomId');
+        return;
+      }
+
+      final result = await showFutureLoadingDialog<String>(
+        context: context,
+        future: () async {
+          try {
+            // Retrieve participating servers
+            final servers = roomAliasResult.servers ?? ['staging.tawkie.fr'];
```

Comment:
This probably shouldn't end up in production code. Try checking whether the server list is empty first.